### PR TITLE
Update parameters used for snow cover fraction scheme

### PR DESCRIPTION
**Pull request by spencerkclark**
**Description**

This PR updates some parameters used in the snow cover fraction scheme of Noah-MP to be consistent with those in official version 4.5.  The commit that added these in the official Noah-MP repository is [here](https://github.com/NCAR/noahmp/commit/03690c5a2df0b6e4352f358f9dab5345c7bb1239).

cc: @kaiyuan-cheng @lharris4

**How Has This Been Tested?**

I have tested this in a year-long C24 resolution simulation with prescribed observed 2020 sea surface temperature and sea ice and compared the monthly mean surface albedo to ERA5.  Quantitatively in terms of land RMSE the differences are marginal, but it does seem to help the albedo bias be less systematically negative.

#### Before

2024-06-17-albedo-bias-SHiELD-Noah-MP (soil color, iopt_alb = 3, iopt_snf = 3)

#### After

2024-06-17-albedo-bias-SHiELD-Noah-MP (soil color, iopt_alb = 3, iopt_snf = 3, scffac)

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by lharris4 — APPROVED** [review]

**Comment by lharris4** [comment]
I think this looks OK. Thanks.

**Comment by spencerkclark** [comment]
Thanks @laurenchilutti and @lharris4!  Yes, it would be great to get @kaiyuan-cheng's approval before merging—it should hopefully be straightforward to review.

**Review by kaiyuan-cheng — APPROVED** [review]
It looks good.

**Comment by spencerkclark** [comment]
Perfect, thanks @kaiyuan-cheng!

**Pull request merged**
